lib/Controller.js

```diff
@@ -70,8 +72,17 @@ module.exports = (function(){
         var that = this;
         var liveEntry = new LiveEntryCtor(e);
 
+        this.crossDCsStreamPersistence.removeEntry(e.entryId);
+
         return liveEntry.start()
-            .then(function(){
+            .then(function () {
+                return liveEntry;
+            })
+            .then((liveEntry) => {
+                that.crossDCsStreamPersistence.flushEntryList()
```

Comment:
why do we call flushEntry list here? the remove Entry that you call few lines ago should do it
the logic of the crossDCsStreamPersistence should not be handled from ourside